### ISSUE TITLE
Feature Request: Allow for sloppy=True in TFRecordDataset's init

### ISSUE THREAD

**Issue by GPhilo** [tensorflow/tensorflow#18827]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: -
- **TensorFlow installed from (source or binary)**: -
- **TensorFlow version (use command below)**: 1.8.0 rc1
- **Python version**: -
- **Bazel version (if compiling from source)**:-
- **GCC/Compiler version (if compiling from source)**: -
- **CUDA/cuDNN version**: -
- **GPU model and memory**: -
- **Exact command to reproduce**: -

### Describe the problem
From TF 1.8 `TFRecordDataset` has a new init parameter `num_parallel_reads` which essentially wraps a `ParallelInterleaveDataset` around the `_TFRecordDataset` instance we're reading from.
In the current implementation, the parallel interleave is hardcoded with `sloppy=False`.
Would it be possible to add `sloppy` as an additional parameter to TFRecordDataset? That would allow to simplify this:
```python
files = tf.data.list_files('somewhere/*')
dataset = files.apply(tf.data.parallel_interleave(TFRecordDataset, cycle_length=num_parallel_calls, sloppy=True))
```
to
```python
files = tf.data.list_files('somewhere/*')
dataset = TFRecordDataset(files, num_parallel_calls=num_parallel_calls, sloppy=True)
```
(Note that the underlying implementations are identical)

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 46 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 61 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 76 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 91 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 106 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 121 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 136 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 151 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 166 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mrry** [comment]
After giving this request some thought, we have decided *not* to add an extra argument to the `tf.data.TFRecordDataset` initializer. However, we recently added the `tf.data.Dataset.with_options()` method, and it seems appropriate to add an option to `tf.data.Options` that will make it possible to turn on "sloppiness" for an entire pipeline.

Assigning this to @jsimsa, who already has the corresponding internal bug assigned to him!
